tests/test_ilastik/test_applets/dataSelection/testOpDataSelection.py

```diff
@@ -658,198 +662,258 @@ def test_load_single_file_with_list(self):
+                "voxel_offset": [0, 0, 0],
+                "chunk_sizes": [list(CHUNK_SIZE_XYZ)],
+                "encoding": "raw",
+            },
         ],
+    }
+
+    def test_load_precomputed_chunks_over_http(self, graph, monkeypatch):
+        _ = MockRemoteDataset(monkeypatch, self.MOCK_DATASET_URL, self.INFO_JSON, self.CHUNKS)
+        op = OpDataSelection(graph=graph)
+        op.WorkingDirectory.setValue(os.getcwd())
+        op.ActiveScale.setValue(0)
+        op.Dataset.setValue(MultiscaleUrlDatasetInfo(url=self.MOCK_DATASET_URL))
+        loaded_scale0 = op.Image[:].wait()
+        assert numpy.allclose(loaded_scale0, self.IMAGE_SCALED)
+
+        op.ActiveScale.setValue(1)
+        loaded_scale1 = op.Image[:].wait()
+        assert numpy.allclose(loaded_scale1, self.IMAGE_ORIGINAL)
+
+
+class TestOpDataSelection_DatasetInfo:
+    MOCK_PRECOMPUTED_URL = "precomputed://https://mocked.com/precomputed_dataset"
+    MOCK_PRECOMPUTED_INFO = {
+        "type": "image",
+        "data_type": "uint16",
+        "num_channels": 1,
+        "scales": [{"chunk_sizes": [[1, 1, 1]], "resolution": [""], "size": [1, 1, 1]}],
+    }
+
+    @pytest.fixture
+    def mock_project(self, data_path):
+        data = mock.Mock()
+        data.shape = (10, 10)
+        data.attrs = {}
+        project = mock.MagicMock()
+        project.filename = data_path / "mock.file"
+        project.__getitem__.return_value = data
+        return project
+
+    @pytest.mark.parametrize(
+        "info_class, expected_sub_path",
         [
-            "rgb*1c",
-            ".png",
-            "t",
-            numpy.stack([_data["rgb00c"], _data["rgb10c"], _data["rgb20c"]], axis=0)[..., None] * 255,
+            (ProjectInternalDatasetInfo, ""),
+            (FilesystemDatasetInfo, "inputdata"),
+            (RelativeFilesystemDatasetInfo, "inputdata"),
+            (MultiscaleUrlDatasetInfo, ""),
         ],
-        ["rgb*3c", ".h5/data", "t", numpy.stack([_data["rgb03c"], _data["rgb13c"], _data["rgb23c"]], axis=0)],
-        ["rgb*3c", ".png", "t", numpy.stack([_data["rgb03c"], _data["rgb13c"], _data["rgb23c"]], axis=0) * 255],
-        ["rgb*3c", ".tiff", "t", numpy.stack([_data["rgb03c"], _data["rgb13c"], _data["rgb23c"]], axis=0) * 255],
-    ],
-)
-def test_stack_along(test_data_dir, graph, name, extension, sequence_axis, expected):
-    fileName = test_data_dir / f"{name}{extension}"
-    reader = OpDataSelection(graph=graph, forceAxisOrder=False)
-    reader.WorkingDirectory.setValue(os.getcwd())
-    reader.Dataset.setValue(FilesystemDatasetInfo(filePath=str(fileName), sequence_axis=sequence_axis))
-    read = reader.Image[...].wait()
-
-    assert numpy.allclose(read, expected), f"{name}: {read.shape}, {expected.shape}"
+    )
+    def test_default_export_paths(self, data_path, mock_project, monkeypatch, info_class, expected_sub_path):
+        _ = MockRemoteDataset(monkeypatch, self.MOCK_PRECOMPUTED_URL, self.MOCK_PRECOMPUTED_INFO, {})
```

Comment:
It's necessary because the same way that File-based datasets read the file header to get metadata, the Url-based dataset requests the Url to get metadata. I agree that this probably shouldn't be happening during DatasetInfo instantiation already, but so it currently is... I added a comment to clarify.